ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -968,6 +979,9 @@ private boolean shouldResumePreviousDump(Path lastDumpPath, boolean isBootStrap)
       return false;
     }
     Path hiveDumpPath = new Path(lastDumpPath, ReplUtils.REPL_HIVE_BASE_DIR);
+    if (tableExpressionModified(new DumpMetaData(hiveDumpPath, conf))) {
```

Comment:
We can  reuse the  same DumpMetaData object in the below check also, would avoid FileSystem read call:
    if (isBootStrap) {
      return shouldResumePreviousDump(new DumpMetaData(hiveDumpPath, conf));
    }
Second thing, I was also wondering if isBootStrap is true , do we even need that check or we can simply return false.

Reply:
Yes we can reuse the same.
If its bootstrap then also we need this check. Because for bootstrap, we reuse the same directory, delete the metadata dir, but data dir is still reused.